test/TestProjects/TypeSchemaMapping/Customized/SecondModel.cs

```diff
@@ -0,0 +1,20 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System.Collections.Generic;
+using Azure.Core;
+
+namespace TypeSchemaMapping.Models
+{
+    // This model would map without the attribute because the name and namespace is the same.
+    internal partial class SecondModel
```

Comment:
Example of how mapping just works when the full type name matches.

Reply:
How do I work around this when I want to have my own `SecondModel` class that's pretty different than the generated `SecondModelInternal`?  Does putting `[CodeGenModel("SecondModel")] internal class SecondModelInternal { ... }` tell the generator not to merge into my own `public class SecondModel { ... }`?

Reply:
> Does putting [CodeGenModel("SecondModel")] internal class SecondModelInternal { ... } tell the generator not to merge into my own public class SecondModel { ... }?

Yes, remapping takes precedence over trying to map to existing types.